admin/class-my-yoast-route.php

```diff
@@ -0,0 +1,198 @@
+		$action = $this->get_action();
+		switch ( $action ) {
+			case 'connect';
+				$this->connect();
+			break;
+		}
+
+		$this->stop_execution();
+	}
+
+	/**
+	 * Checks if the current page is the MyYoast route.
+	 *
+	 * @param string $route The myyoast route.
+	 *
+	 * @return bool True when url is the myyoast route.
+	 */
+	protected function is_myyoast_route( $route ) {
+		return ( $route === self::PAGE_IDENTIFIER );
+	}
+
+	/**
+	 * Compares an action to a list of allowed actions to see if it is valid.
+	 *
+	 * @param string $action The action to check.
+	 *
+	 * @return bool True if the action is valid.
+	 **/
+	protected function is_valid_action( $action ) {
+		$allowed_actions = array( 'connect' );
+
+		return in_array( $action, $allowed_actions );
+	}
+
+	/**
+	 * Connects to MyYoast, generates a ClientId if needed.
+	 *
+	 * @return void
+	 */
+	protected function connect() {
+		$config    = $this->get_client()->get_configuration();
+		$client_id = $config['clientId'];
+
+		if ( empty( $client_id ) ) {
+			$client_id = $this->generate_userid();
+
+			$this->get_client()->save_configuration(
+				array(
+					'clientId' => $client_id,
+				)
+			);
+		}
+
+		$this->redirect(
+			'https://my.yoast.com/connect',
+			array(
+				'url'          => WPSEO_Utils::get_home_url(),
+				'client_id'    => $client_id,
+				'extensions'   => array(),
+				'redirect_url' => admin_url( 'admin.php?page=' . WPSEO_Admin::PAGE_IDENTIFIER ),
+			)
+		);
+	}
+
+	/**
+	 * Creates a new MyYoast Client instance.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return WPSEO_MyYoast_Client Instance of the myyoast client.
+	 */
+	protected function get_client() {
+		static $client;
+
+		if ( ! $client ) {
+			$client = new WPSEO_MyYoast_Client();
+		}
+
+		return $client;
+	}
+
+	/**
+	 * Abstracts the action from the url.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return string The action from the url.
+	 */
+	protected function get_action() {
+		return filter_input( INPUT_GET, 'action' );
+	}
+
+	/**
+	 * Generates an URL-encoded query string, redirects there.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @param string $url        The url to redirect to.
+	 * @param array  $query_args The additional arguments to build the url from.
+	 *
+	 * @return void
+	 */
+	protected function redirect( $url, $query_args ) {
+		wp_redirect( $url . '?' . http_build_query( $query_args ) );
+		exit;
+	}
+
+	/**
+	 * Checks if current user is allowed to access the route.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return bool True when current user has rights to manage options.
+	 */
+	protected function can_access_route() {
+		return current_user_can( 'wpseo_manage_options' );
+	}
+
+	/**
+	 * Stops the execution.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return void;
+	 */
+	protected function stop_execution() {
+		exit;
+	}
+
+	/**
+	 * Generates an unique user id.
+	 *
+	 * @codeCoverageIgnore
+	 *
+	 * @return string The generated userid.
+	 */
+	protected function generate_userid() {
```

Comment:
Would rename this to `generate_uuid`. `userid` implies it's for users which generally have integers as IDs.